### ISSUE TITLE
Inventory fails with bucket ACLs permission errors for Composite Root

### ISSUE THREAD

**Issue by marmolejo** [forseti-security/forseti-security#2827]
v2.14.1 constantly fails in the inventory stage by not being able to get ACLs for buckets:

```
2019-05-10 08:59:16,892 ERROR [forseti-security][2.14.1] google.cloud.forseti.common.gcp_api.storage(get_bucket_acls): GCP API Error: unable to get bucketAccessControls, bucket = XXXX, from GCP:
<HttpError 403 when requesting https://www.googleapis.com/storage/v1/b/XXXX/acl?alt=json returned "forseti-server-gcp-account@project.iam.gserviceaccount.com does not have storage.buckets.get access to XXXX.">
{
 "error": {
  "errors": [
   {
    "domain": "global",
    "reason": "forbidden",
    "message": "forseti-server-gcp-account@project.iam.gserviceaccount.com does not have storage.buckets.get access to XXXX."
   }
  ],
  "code": 403,
  "message": "forseti-server-gcp-account@project.iam.gserviceaccount.com does not have storage.buckets.get access to XXXX."
 }
}
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/forseti_security-2.14.1-py2.7.egg/google/cloud/forseti/common/gcp_api/storage.py", line 523, in get_bucket_acls
    'items')
  File "/usr/local/lib/python2.7/dist-packages/forseti_security-2.14.1-py2.7.egg/google/cloud/forseti/common/gcp_api/api_helpers.py", line 87, in flatten_list_results
    for page in paged_results:
  File "/usr/local/lib/python2.7/dist-packages/forseti_security-2.14.1-py2.7.egg/google/cloud/forseti/common/gcp_api/repository_mixins.py", line 59, in list
    yield self.execute_query(verb=verb, verb_arguments=arguments)
  File "/usr/local/lib/python2.7/dist-packages/forseti_security-2.14.1-py2.7.egg/google/cloud/forseti/common/gcp_api/_base_repository.py", line 506, in execute_query
    return self._execute(request)
  File "/usr/local/lib/python2.7/dist-packages/forseti_security-2.14.1-py2.7.egg/google/cloud/forseti/common/util/replay.py", line 168, in replay_wrapper
    return f(self, request, *args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/forseti_security-2.14.1-py2.7.egg/google/cloud/forseti/common/util/replay.py", line 81, in record_wrapper
    return f(self, request, *args, **kwargs)
  File "build/bdist.linux-x86_64/egg/retrying.py", line 49, in wrapped_f
    return Retrying(*dargs, **dkw).call(f, *args, **kw)
  File "build/bdist.linux-x86_64/egg/retrying.py", line 206, in call
    return attempt.get(self._wrap_exception)
  File "build/bdist.linux-x86_64/egg/retrying.py", line 247, in get
    six.reraise(self.value[0], self.value[1], self.value[2])
  File "build/bdist.linux-x86_64/egg/retrying.py", line 200, in call
    attempt = Attempt(fn(*args, **kwargs), attempt_number, False)
  File "/usr/local/lib/python2.7/dist-packages/forseti_security-2.14.1-py2.7.egg/google/cloud/forseti/common/gcp_api/_base_repository.py", line 530, in _execute
    num_retries=self._num_retries)
  File "/home/ubuntu/forseti-security/.eggs/google_api_python_client-1.7.7-py2.7.egg/googleapiclient/_helpers.py", line 130, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/home/ubuntu/forseti-security/.eggs/google_api_python_client-1.7.7-py2.7.egg/googleapiclient/http.py", line 851, in execute
    raise HttpError(resp, content, uri=self.uri)
HttpError: <HttpError 403 when requesting https://www.googleapis.com/storage/v1/b/XXXX/acl?alt=json returned "forseti-server-gcp-account@project.iam.gserviceaccount.com does not have storage.buckets.get access to XXXX.">
2019-05-10 08:59:16,893 WARNING [forseti-security][2.14.1] google.cloud.forseti.services.inventory.base.resources(get_gcs_policy): Could not get bucket Access Control policy: GCP API Error: unable to get bucketAccessControls, bucket = XXXX, from GCP:
<HttpError 403 when requesting https://www.googleapis.com/storage/v1/b/XXXX/acl?alt=json returned "forseti-server-gcp-account@project.iam.gserviceaccount.com does not have storage.buckets.get access to XXXX.">
{
 "error": {
  "errors": [
   {
    "domain": "global",
    "reason": "forbidden",
    "message": "forseti-server-gcp-account@project.iam.gserviceaccount.com does not have storage.buckets.get access to XXXX."
   }
  ],
  "code": 403,
  "message": "forseti-server-gcp-account@project.iam.gserviceaccount.com does not have storage.buckets.get access to XXXX."
 }
}
```

**Comment by gkowalski-google** [comment]
Hi @marmolejo, are you still experiencing this issue or were you able to get it resolved?

**Comment by gkowalski-google** [comment]
@marmolejo I am closing this issue since we have not heard back from you. Please re-open if you need some help in regards to this. Thanks!

**Comment by marmolejo** [comment]
Sorry, I was out of office for long time. This bug is still happening as of version 2.24. When inventory stage tries to get bucketAccessControls, it gets this error. Apparently the service account running as the server doesn't have the storage.buckets.get permission.

This is a bit tricky because there is no role that only has this permission. Only roles/storage.legacyBucketReader has storage.buckets.get and storage.objects.list, which would be quite ideal for this but unfortunately it cannot be set for service accounts only to buckets:

https://cloud.google.com/storage/docs/access-control/iam-roles#legacy-roles

The only role that has this permission is roles/storage.admin, but unfortunately is to broad to be applied to forseti. So the only option I see here is to create a custom role that has this permission.

**Comment by gkowalski-google** [comment]
@marmolejo Are you using the composite root resources?

**Comment by marmolejo** [comment]
Yes.

**Comment by gkowalski-google** [comment]
This might be hard to automate, if so, then look into adding some documentation to the website for additional permissions/roles that are necessary to support Composite Root.

**Comment by gkowalski-google** [comment]
I believe this is a dupe of #2827, if so, can you consolidate and close one.

**Comment by red2k18** [comment]
@hshin-g I see the documentation has been updated with the custom role. Is there any remaining work on this issue or it can be closed?

**Comment by hshin-g** [comment]
I think @gkowalski-google said he was seeing this error while he was not running on composite root, but I can't seem to reproduce it on my end, so will defer to @gkowalski-google on this one.

**Comment by red2k18** [comment]
@gkowalski-google Is there anything else we need to do to fix the inventory with bucket ACLs permission errors for composite root? If so, please list here.

Are you still seeing this error on non-composite root? If so, I will open another ticket to track that.

**Comment by gkowalski-google** [comment]
@marmolejo We are going to close out this issue. To resolve it, create a custom role and assign the storage.buckets.get permission, and assign this role to the server SA. This has been added to the website docs which will be published with the Forseti v2.25.0 release.
